util/collections/machine_filters_test.go

```diff
@@ -454,6 +454,100 @@ func TestHasNode(t *testing.T) {
 	})
 }
 
+func TestHasUnhealthyControlPlaneComponentCondition(t *testing.T) {
+	t.Run("nil machine returns true", func(t *testing.T) {
```

Comment:
As all other of these funcs return false in this case, I think it would make sense to also return false here

Reply:
sure, updated per comment.